### PR TITLE
Add state tracking for unicode

### PR DESCRIPTION
**Pull request by nanjekyejoannah**
Just state, warning already existed. 
Format methods are shared, so warnings are covered in #38

### CODE REVIEW

**Comment by ltratt** [comment]
@nanjekyejoannah This one failed with errors --- worth fixing?

**Comment by nanjekyejoannah** [comment]
I added a missing attribute and fixed a test.

File: Objects/unicodeobject.c [hunk 1]
```diff
@@ -1211,17 +1211,29 @@ PyObject *PyUnicode_FromEncodedObject(register PyObject *obj,
 #else
 #endif
     
+    // if (PyUnicode_Check(obj)) {
```
Comment (ltratt) [inline Objects/unicodeobject.c]: Should we just delete this chunk?

**Comment by nanjekyejoannah** [comment]
Let me first run the whole suite locally myself too.
I usually just run the affected tests.

**Comment by nanjekyejoannah** [comment]
Fixing this bug, uncovered that it is related to us adding `bstate` on object, we break many things but we can discuss the compromise tommorrow.

**Comment by nanjekyejoannah** [comment]
Lets figure out [tracking](https://github.com/softdevteam/pygrate2/pull/41) in before we get back here

**Pull request closed without merging**